datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -121,39 +136,182 @@ def read_dask(
 
         import dask.dataframe as dd
 
-        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id)
         return dd.read_parquet(
             path,
             columns=read_columns,
-            filters=filters if filters and len(filters) else None,
+            filters=filters,
             engine=kwargs.get("engine", "pyarrow"),
             **kwargs,
         )
 
     def read_spark(
-        self, columns: Optional[str] = None, run_id: Optional[str] = None, conf=None, **kwargs
-    ) -> "pyspark.sql.DataFrame":
+        self,
+        columns: Optional[str] = None,
+        run_id: Optional[str] = None,
+        conf: Optional["SparkConf"] = None,
+        storage_format: str = "parquet",
+        **kwargs,
+    ) -> "SparkDataFrame":
         if not (self.mode & Mode.READ):
```

Comment:
is it now assumed that you must be in mode.READ to read the data and mode.WRITE to write the data? or is there a READ_WRITE? I remember that being a change being considered but not seeing it..

Reply:
^^ to Ashley's comment, seems this should be: `if self.mode and self.mode not in [Mode.READ, Mode.READ_WRITE]:`

Reply:
bitwise operator :D all good